### PR TITLE
fix type evaluating for `mult` and `div` arithmetic operations

### PR DESCRIPTION
**Pull request by jennifersp**
- For `Mult` expression, the `.Type()` method is fixed that it returns correct type for decimal type results, which uses the sum of precision and scale values of its right and left values. 
- For `Div` expression, the `.Type()` method is fixed that it returns the correct precision and scale for the final result.

### CODE REVIEW

File: sql/expression/plus.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+func (p *Plus) IsNullable() bool {
+	if types.IsDatetimeType(p.Type()) || types.IsTimestampType(p.Type()) {
+		return true
+	}
+
+	return p.BinaryExpression.IsNullable()
+}
+
+// Type returns the greatest type for given operation.
+func (p *Plus) Type() sql.Type {
```
Comment (fulghum) [inline sql/expression/plus.go:79]: This function looks like it's the same as the `Type()` function for the new `Minus` type, too. It would be helpful to have them reuse the same helper function, instead of having this code copied, so that when we fix any issues in one implementation, we don't have to remember to fix it in a second spot.

**Review by fulghum — COMMENTED** [review]
Code looks correct, but there's too much copy/paste code in here. 

With a little more refactoring, you could probably get rid of a couple hundred lines of copy/paste. For example, if you created a new `ArithmeticOpBase` struct it could give all of the arithemtic types implementations for `LeftChild()`, `RightChild()`, `SetOpCount(int32)`, `String()`, `DebugString()`, `IsNullable()`, `CollationCoercibility(ctx)`, and possible some more, too. `Type()` also looks like it's the same for Plus/Minus/Mult, so that would get rid of a lot of copy/paste code, too. Totally fine for `Div` to have it's own separate implementation of course.

**Comment by jennifersp** [comment]
Decided **not** to refactor `Arithmetic` split into separate expressions for plus, minus and mult.

File: enginetest/queries/queries.go [hunk 2]
```diff
@@ -4612,6 +4612,14 @@ Select * from (
 			{uint64(18446744073709551613)},
 		},
 	},
+	{
+		Query:    "select 0 in (1 / 1000), 0 in (1 / 1000, 0), 0.001 in (1 / 1000, 0), 0.0001 in (1 / 1000, 0);",
```
Comment (jycor) [inline enginetest/queries/queries.go:4616]: This stops working when the scale is greater than 4
Dolt:
```sql
tmp> select 0 in (1/100000);
+-----------------+
| 0 in (1/100000) |
+-----------------+
| true            |
+-----------------+
1 row in set (0.00 sec)
```

MySQL
```sql
mysql> select 0 in (1/100000);
+-----------------+
| 0 in (1/100000) |
+-----------------+
|               0 |
+-----------------+
1 row in set (0.0004 sec)
```

However, MySQL does start to break around `100 000 000 000` (scale 11)
```
mysql> select 0 in (1/100000000000);
+-----------------------+
| 0 in (1/100000000000) |
+-----------------------+
|                     1 |
+-----------------------+
1 row in set (0.0004 sec)

mysql> select 0 = (1/100000000000);
+----------------------+
| 0 = (1/100000000000) |
+----------------------+
|                    1 |
+----------------------+
1 row in set (0.0004 sec)
```

It looks like MySQL has some other internal logic dealing with decimal precision.
I think this should be dealt with in another PR; a skipped test or issue is good enough for now.

File: sql/expression/arithmetic.go [hunk 3]
```diff
@@ -511,6 +515,32 @@ func minus(lval, rval interface{}) (interface{}, error) {
...
+		return types.Float64
+	}
+
+	lPrec := lType.(types.DecimalType_).Precision()
+	lScale := lType.(types.DecimalType_).Scale()
+	rPrec := rType.(types.DecimalType_).Precision()
+	rScale := rType.(types.DecimalType_).Scale()
+
+	maxWhole := (lPrec - lScale) + (rPrec - rScale)
+	maxFrac := lScale + rScale
```
Comment (jycor) [inline sql/expression/arithmetic.go:534]: Can you call this `maxScale` instead

**Review by jycor — APPROVED** [review]
LGTM, small nit with naming.

File: enginetest/queries/queries.go (hunk 2 above)
Comment (jennifersp) [reply enginetest/queries/queries.go:4616]: tracking issue: https://github.com/dolthub/dolt/issues/7207
